conn.go

```diff
@@ -127,7 +127,6 @@ func (c *Conn) listenResponse(id uint32) (*Header, error) {
 	defer func() {
 		c.Lock()
 		close(c.listeners[id])
-		delete(c.listeners, id)
```

Comment:
I added it back, and get around the data race by not using map[key] over and over again, since we already do value, ok := map[key], we can just use the value, instead of reading map.